Sources/ArgumentParser/Parsable Types/ParsableArgumentsValidation.swift

```diff
@@ -185,3 +186,63 @@ struct ParsableArgumentsCodingKeyValidator: ParsableArgumentsValidator {
     }
   }
 }
+
+/// Ensure argument names are unique within a `ParsableArguments` or `ParsableCommand`.
+struct ParsableArgumentsUniqueNamesValidator: ParsableArgumentsValidator {
+  struct Error: Swift.Error, CustomStringConvertible {
+    var duplicateNames: [String: Int] = [:]
+
+    var occurred: Bool {
+      !duplicateNames.isEmpty
+    }
+
+    var description: String {
+      var description = duplicateNames.reduce(into: "") { description, entry in
```

Comment:
You could write this as `map` and then `joined` to avoid the `removeLast` after building the string.

Reply:
Thank you for spotting this. The `reduce` was a remnant from when this error type was more complicated, but after I simplified it I didn't rethink this part.

That extra complication is something I wanted to discuss. This `Error` used to be an enum that also represented the case where the user defined an option or flag named `"h"` or `"help"`, in which case the built-in help is masked. For example,
```
import ArgumentParser

struct RedefinedHelp: ParsableCommand {
    @Flag()
    var help: Bool

    func run() throws {
        print("help: \(help)")
    }
}

RedefinedHelp.main(["--help"])
```
gives
```
help: true
```
But I had problems detecting this reliably in the validator, so I removed that code. But now I think I might know what the problem was, so I'll give it another try, if you think detecting this case is a good idea.

Reply:
I pushed a commit to simplify using `map`, as suggested. Much nicer! I haven't yet revisited the idea of detecting the case where the user defines their own `help` flag or option. I may have a chance to do that today.

Reply:
That case is a clean override, and supported, so we wouldn't want to error there. The user-defined flag/option should always win, so please file a bug if you see the other behavior!